### PR TITLE
OCPBUGS#37036: Fix `opm` base image

### PR DESCRIPTION
**Pull request by michaelryanpeter**
Version(s): 4.15+

Issue: [OCPBUGS-37036](https://issues.redhat.com/browse/OCPBUGS-37036)

Link to docs preview: https://79646--ocpdocs-pr.netlify.app/openshift-enterprise/latest/operators/admin/olm-managing-custom-catalogs.html

QE review:
- [x] QE has approved this change.

Additional information:

### CODE REVIEW

**Review by grokspawn — APPROVED** [review]

**Comment by michaelryanpeter** [comment]
@jianzhangbjz PTAL

File: modules/olm-creating-fb-catalog-image.adoc [hunk 1]
```diff
@@ -6,7 +6,7 @@ ifdef::openshift-origin[]
 :registry-image: quay.io/operator-framework/opm:latest
 endif::[]
 ifndef::openshift-origin[]
-:registry-image: registry.redhat.io/openshift4/ose-operator-registry:v{product-version}
+:registry-image: registry.redhat.io/openshift4/ose-operator-registry-rhel9:v{product-version}
```
Comment (dfitzmau) [inline modules/olm-creating-fb-catalog-image.adoc:9]: I'm assuming "{product-version}" will change to the version number and not Branch Build. Would be worth checking on Pantheon. 

Screenshot from 2024-07-30 15-25-21

**Review by dfitzmau — APPROVED** [review]
/remove-label peer-review-in-progress

/label peer-review-done

**Pull request merged**